p2p/peer_tracker.go

```diff
@@ -180,12 +188,40 @@ func (p *peerTracker) gc() {
 				}
 			}
 			p.peerLk.Unlock()
+
+			p.dumpPeers()
 		}
 	}
 }
 
+// dumpPeers stores peers to the PeerTracker's PeerIDStore if
+// it exists.
+func (p *PeerTracker) dumpPeers() {
+	if p.pidstore == nil {
+		return
+	}
+
+	peers := make([]peer.ID, 0, len(p.trackedPeers))
+
+	p.peerLk.RLock()
+	for id := range p.trackedPeers {
+		peers = append(peers, id)
+	}
+	p.peerLk.RUnlock()
```

Comment:
If you are going to pre-allocate the slice, then in the for loop you should use the index. Otherwise you are reallocating memory every time you use `append`.

```suggestion
	peers := make([]peer.ID, 0, len(p.trackedPeers))

	p.peerLk.RLock()
	for i, id := range p.trackedPeers {
		peers = peers[i] = id
	}
	p.peerLk.RUnlock()
```

Reply:
The slice has preallocated `cap`, not `len`, so the code is correct.

Reply:
I've already allocated the max size for the slice - so append will not cause a realloc here.